tests/integration/backward_compatible/ssl_tests/hostname_verification/ssl_hostname_verification_test.py

```diff
@@ -0,0 +1,134 @@
+from hazelcast.config import SSLProtocol
+from hazelcast.errors import IllegalStateError
+from tests.base import HazelcastTestCase
+from tests.util import compare_client_version, get_abs_path
+
+current_directory = os.path.dirname(__file__)
+
+MEMBER_CONFIG = """
+<hazelcast xmlns="http://www.hazelcast.com/schema/config"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://www.hazelcast.com/schema/config
+           http://www.hazelcast.com/schema/config/hazelcast-config-5.0.xsd">
+    <network>
+        <ssl enabled="true">
+            <factory-class-name>
+                com.hazelcast.nio.ssl.BasicSSLContextFactory
+            </factory-class-name>
+            <properties>
+                <property name="keyStore">%s</property>
+                <property name="keyStorePassword">123456</property>
+                <property name="keyStoreType">PKCS12</property>
+                <property name="protocol">TLSv1.2</property>
+            </properties>
+        </ssl>
+    </network>
+</hazelcast>
+"""
+
+
+@unittest.skipIf(
+    sys.version_info < (3, 7),
+    "Hostname verification feature requires Python 3.7+",
+)
+@unittest.skipIf(
+    compare_client_version("5.1") < 0,
+    "Tests the features added in 5.1 version of the client",
+)
+@pytest.mark.enterprise
+class SslHostnameVerificationTest(HazelcastTestCase):
+    def setUp(self):
+        self.rc = self.create_rc()
+        self.cluster = None
+
+    def tearDown(self):
+        self.shutdown_all_clients()
+        self.rc.terminateCluster(self.cluster.id)
+        self.rc.exit()
+
+    def test_hostname_verification_with_loopback_san(self):
+        # SAN entry is present with different possible values
+        self.start_member_with("tls-host-loopback-san.p12")
+
+        self.start_client_with("tls-host-loopback-san.pem", "127.0.0.1:5701")
+        self.start_client_with("tls-host-loopback-san.pem", "localhost:5701")
+
+    def test_hostname_verification_with_loopback_dns_san(self):
+        # SAN entry is present, but only with `dns:localhost`
+        self.start_member_with("tls-host-loopback-san-dns.p12")
+        self.start_client_with("tls-host-loopback-san-dns.pem", "localhost:5701")
+
+        with self.assertRaisesRegex(IllegalStateError, "Unable to connect to any cluster"):
+            self.start_client_with("tls-host-loopback-san-dns.pem", "127.0.0.1:5701")
+
+    def test_hostname_verification_with_different_san(self):
+        # There is a valid entry, but it does not match with the address of the member.
+        self.start_member_with("tls-host-not-our-san.p12")
+
+        with self.assertRaisesRegex(IllegalStateError, "Unable to connect to any cluster"):
+            self.start_client_with("tls-host-not-our-san.pem", "localhost:5701")
+
+        with self.assertRaisesRegex(IllegalStateError, "Unable to connect to any cluster"):
+            self.start_client_with("tls-host-loopback-san-dns.pem", "127.0.0.1:5701")
```

Comment:
Did you do a copy paste mistake here? 
```suggestion
            self.start_client_with("tls-host-not-our-san.pem", "127.0.0.1:5701")
```

Reply:
Yes, thanks for catching this. I have updated the tests to re-use the file name to avoid such mistakes now.